src/gui/FolderColumnView.ts

```diff
@@ -19,26 +19,37 @@ export type Attrs = {
 
 export class FolderColumnView implements Component<Attrs> {
 	view({ attrs }: Vnode<Attrs>): Children {
-		return m(".flex.height-100p.nav-bg", [
-			m(DrawerMenu, attrs.drawer),
-			m(".folder-column.flex-grow.overflow-x-hidden.flex.col", landmarkAttrs(AriaLandmarks.Navigation, lang.getMaybeLazy(attrs.ariaLabel)), [
-				this.renderMainButton(attrs),
-				m(
-					".scroll.scrollbar-gutter-stable-or-fallback.visible-scrollbar.overflow-x-hidden.flex.col.flex-grow",
-					{
-						onscroll: (e: Event) => {
-							const target = e.target as HTMLElement
-							if (attrs.button == null || target.scrollTop === 0) {
-								target.style.borderTop = ""
-							} else {
-								target.style.borderTop = `1px solid ${theme.content_border}`
-							}
+		return m(
+			".flex.height-100p.nav-bg",
+			{
+				onupdate(vnode: m.VnodeDOM<Attrs>): any {
+					if (vnode.dom.parentElement) {
+						const trapFocus = vnode.dom.parentElement.style.visibility === "visible"
+						handleFocus(trapFocus, ["nav", ".view-columns"])
+					}
+				},
```

Comment:
This couples the `FolderColumnView` to the `ViewSlider` which will be not so nice when we refactor the `ViewSlider`. I would also advise against giving components access to their parents if possible.

Have you tried creating an `isFocused` attribute then passing in `(vnode.dom as HTMLElement).style.visibility === "visible"` from the `ViewSlider` into it?